language-server/langserver/src/main/java/edu/umn/cs/melt/silver/langserver/Util.java

```diff
@@ -119,4 +128,71 @@ public FileVisitResult visitFile(Path file, BasicFileAttributes attrs) throws IO
             }
         });
     }
+
+    /**
+     * Dynamically load a Copper parser from a compiled Silver grammar.
+     * 
+     * @param <ROOT> The start nonterminal type of the parser
+     * @param jarPath The path to the jar file to be loaded
+     * @param name The name of the declared parser, prefixed by its grammar
+     * @param rootClass The class of the start nonterminal for the parser
+     * @return A factory object for instantiating the parser
+     * @throws SecurityException
+     * @throws ReflectiveOperationException
+     */
+    @SuppressWarnings("unchecked")
+    public static <ROOT> Supplier<SilverCopperParser<ROOT>> loadCopperParserFactory(
+        final Path jarPath, final String name, final Class<ROOT> rootClass)
+        throws SecurityException, ReflectiveOperationException {
+        // Initialize a class loader for the jar
+        URLClassLoader loader;
+        try {
+            loader = new URLClassLoader(new URL[] {jarPath.toUri().toURL()}, Util.class.getClassLoader());
+        } catch (MalformedURLException e) {
+            throw new RuntimeException(e);
+        }
+
+        // Load the parser class
+        String pkg = String.join(".", name.substring(0, Math.max(name.lastIndexOf(":"), 0)).split(":"));
+        String parserClassName = pkg + ".Parser_" + String.join("_", name.split(":"));
+        Class<?> parserClass = Class.forName(parserClassName, true, loader);
+
+        // Sanity check: make sure it's actually a Silver-declared Copper parser
+        if (!SilverCopperParser.class.isAssignableFrom(parserClass)) {
+            throw new ReflectiveOperationException("Loaded class is not a Silver-generated Copper parser");
+        }
+
+        // Hacky way of determining the actual start nonterminal type of the parser, from its superclass generic type args.
+        Type[] genericParams = ((ParameterizedType)parserClass.getGenericSuperclass()).getActualTypeArguments();
+        if (genericParams.length != 2) {
+            throw new ReflectiveOperationException("Could not determine parser root type");
```

Comment:
This shouldn't be possible to reach, right? Since the superclass would be CopperParser and we know this from the last check. Is this another sanity check? Either way maybe the exception message could be clearer?

Reply:
Essentially a sanity check.  This could break if there was a change to the super class used by Copper.